### PR TITLE
Add tag name in the commits list

### LINKED ISSUE #10036: [Feature] Git tag support in commits history
- Gitea version (or commit ref): 1.10.3
- Git version: N/A
- Operating system: Linux x86_64
- Database:
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite

## Description

It would great if Gitea can support git tags as well in commits history.

## Screenshots

Current Gitea commits history:
image

[1] Here a visual example of Bitbucket Server commits history, note the tag aside.

image

### PR DESCRIPTION
**Pull request by lunny**
Fix #10036

This PR adds some labels for tags of this commit after the commit message on the commits table. The tag template is share as commit graph's.

Desktop:
image

Mobile:
image

### CODE REVIEW

File: templates/repo/commits_list.tmpl
Comment (delvh) [inline templates/repo/commits_list.tmpl]: I recommend three things about this:
1. extract it into a subtemplate (although I know that one is controversial, some maintainers prefer subtemplates while others prefer a single template)
2. Make it a link that links either to the release or tag
3. Highlight tags with a release in a different color, i.e. releases are colored in the primary color while tags without release are only grey or something like that

Comment (delvh) [reply templates/repo/commits_list.tmpl]: And perhaps 4: Give it only minimal width to display perhaps five chars or so, and display the rest of the name in a tooltip.

Comment (lunny) [reply templates/repo/commits_list.tmpl]: 1,2,3 done. I don't think 4 is necessary because commonly the tag name is not too long.

Comment (delvh) [reply templates/repo/commits_list.tmpl]: The problem is not the `commonly` but exactly the `uncommon`.
Especially on mobile

**Review by delvh — APPROVED** [review]

**Comment by silverwind** [comment]
Do we have other places where we can re-use that new subtemplate? I think commit graph likely has similar tags, but with a border there (I think I like a border on these).

**Comment by lunny** [comment]
> Do we have other places where we can re-use that new subtemplate? I think commit graph likely has similar tags, but with a border there (I think I like a border on these).
aebfe5c604d147be02cac680e174981fb32d7794 use the shared tag name template with commit graph.

**Comment by silverwind** [comment]
Thanks, it does need some style adjustments, the icon makes it too big. Can check later. Also I wonder what the significance of the blue border is.

**Comment by yp05327** [comment]
Should we also show tags in mobile views? It looks not good.

**Comment by silverwind** [comment]
https://github.com/go-gitea/gitea/pull/31160 will rewrite the commits table and make a lot of space for this, so I think we should wait on that to land here.

**Comment by lunny** [comment]
> #31160 will rewrite the commits table and make a lot of space for this, so I think we should wait on that to land here.

I don't think #31160 could be merged at a short time. So I think maybe this can be merged first.

**Review by wolfogre — APPROVED** [review]

File: routers/web/repo/commit.go
Comment (yp05327) [inline routers/web/repo/commit.go]:
```suggestion
		ctx.ServerError("FindTagsByCommitIDs", err)
```

Comment (yp05327) [reply routers/web/repo/commit.go]: And should we break the page if we got error when finding tags?

Comment (lunny) [reply routers/web/repo/commit.go]: fac2eb8038284da4ac5aa8cc34f2841ff2aa781c

File: routers/web/repo/commit.go
Comment (yp05327) [inline routers/web/repo/commit.go]: Locale?

**Review by yp05327 — APPROVED** [review]
I think we can improve the mobile view later.

File: routers/web/repo/commit.go
Comment (lunny) [reply routers/web/repo/commit.go]: 5d81b683f55343f23214efc4b7653ff714ef0d31

File: options/locale/locale_en-US.ini
Comment (wxiaoguang) [inline options/locale/locale_en-US.ini:1277]: Would there be a lot of duplicate (and meaningless) translations?  "Load users failed because of internal error"  "Load branches failed because of internal error" "Load workflows failed because of internal error" "Load releases failed because of internal error" "Load projects failed because of internal error" "Load labels failed because of internal error" "Load milestones failed because of internal error"

Comment (lunny) [reply options/locale/locale_en-US.ini:1277]: I will send another PR to fix all of them.

Comment (lunny) [reply options/locale/locale_en-US.ini:1277]: #31878

**Pull request merged**
